src/rrr/misc/marshal.hpp

```diff
@@ -262,9 +270,9 @@ class Marshal: public NoCopy {
   size_t write_to_fd(int fd);
 
   bookmark *set_bookmark(size_t n);
-  void write_bookmark(bookmark *bm, const void *p) {
+  void write_bookmark(mut_ptr<bookmark>& bm, const void *p) {
```

Comment:
Passing a reference of mut_ptr is not a good idea. It virtually defeats the purpose of using the borrowing model (because you can keep passing the reference of mut_ptr around to have it work as a regular pointer).

Reply:
I will change this to const_ptr

Reply:
Although, I think that even if we pass a mutable reference that is not a violation of the memory safety, it is just that we won't be able to use the same variable until the function that we passed the reference to is executed/returned...